File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/IndexModel.java
package com.raizlabs.android.dbflow.test.sql;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description:
 */
@Table(databaseName = TestDatabase.NAME)
public class IndexModel extends BaseModel {

    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    String name;

    @Column
    long salary;

}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/sql/index/IndexModel.kt
package com.raizlabs.android.dbflow.sql.index

import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.structure.BaseModel
import com.raizlabs.android.dbflow.TestDatabase

/**
 * Description:
 */
@Table(database = TestDatabase::class,
    indexGroups = arrayOf(IndexGroup(name = "customIndex", unique = true)))
class IndexModel : BaseModel() {

    @Column
    @PrimaryKey(autoincrement = true)
    var id: Long = 0

    @Index
    @Column
    var name: String? = null

    @Index
    @Column
    var salary: Long = 0

}
